challenges/2021-06-29-run-of-ones/challenge.md

```diff
@@ -0,0 +1,34 @@
+# Runs of Ones
+
+Given a positive integer `n`, determine the length of the longest run of `1`'s in the binary representation of that integer.
+
+## Business Rules/Errata
+
+- A run of `1`'s is any series of one or more `1`'s in a row, with no `0`'s in between. The first four characters in `11110101` are a run of `1`'s.
+- There's no requirement related to what data type(s) should be used to store the binary representation of `n`.
+- Target time complexity: O(n) (linear)
+- Target space complexity: O(1) (constant)
+
+## Examples
+
+```
+longest_binary_run(156);   // 3
+longest_binary_run(1979);  // 4
+longest_binary_run(2731);  // 2
+longest_binary_run(2185);  // 1
+```
+
+## Tackling This Challenge
```

Comment:
This section includes some references to the challenge it was copied from. The updated Java setup format includes a briefer 'Tackling This Challenge' section with more generic comments.